tests/VerbrauchZeitspanneBaseArchiveTest.php

```diff
@@ -0,0 +1,389 @@
+<?php
+
+declare(strict_types=1);
+
+include_once __DIR__ . '/stubs/GlobalStubs.php';
+include_once __DIR__ . '/stubs/KernelStubs.php';
+include_once __DIR__ . '/stubs/ModuleStubs.php';
+
+use PHPUnit\Framework\TestCase;
+
+class VerbrauchZeitspanneBaseArchiveTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        //Reset
+        IPS\Kernel::reset();
+
+        //Register our core stubs for testing
+        IPS\ModuleLoader::loadLibrary(__DIR__ . '/stubs/CoreStubs/library.json');
+
+        //Register our library we need for testing
+        IPS\ModuleLoader::loadLibrary(__DIR__ . '/../library.json');
+
+        //Register required profiles
+        if (!IPS\ProfileManager::variableProfileExists('~UnixTimestampDate')) {
+            IPS\ProfileManager::createVariableProfile('~UnixTimestampDate', 1);
+        }
+        if (!IPS\ProfileManager::variableProfileExists('~UnixTimestampTime')) {
+            IPS\ProfileManager::createVariableProfile('~UnixTimestampTime', 1);
+        }
+        if (!IPS\ProfileManager::variableProfileExists('~UnixTimestamp')) {
+            IPS\ProfileManager::createVariableProfile('~UnixTimestamp', 1);
+        }
+
+        parent::setUp();
+    }
+
+    public function testDate(): void
+    {
+        $archiveID = IPS_CreateInstance('{43192F0B-135B-4CE7-A0A7-1475603F3060}');
+        $instanceID = IPS_CreateInstance('{F74AA9EF-7B80-4AC8-BE0E-D4C24D8F624B}');
+
+        $sourceVariableID = IPS_CreateVariable(1 /*Integer*/);
+        IPS_SetIdent($sourceVariableID, 'Usage');
+        IPS_SetParent($sourceVariableID, $instanceID);
+
+        IPS_SetProperty($instanceID, 'SourceVariable', $sourceVariableID);
+        IPS_SetProperty($instanceID, 'LevelOfDetail', 0);
+        IPS_ApplyChanges($instanceID);
+        VIZ_SetTime($instanceID, strtotime('5th November 2005 06:00:00'));
+
+        AC_SetLoggingStatus($archiveID, $sourceVariableID, true);
+        $aggregationDataDay = [
+            [
+                'Avg'       => 100,
+                'Duration'  => 60 * 60 * 24,
+                'Max'       => 0,
+                'MaxTime'   => 0,
+                'Min'       => 0,
+                'MinTime'   => 0,
+                'TimeStamp' => strtotime('01-11-2005 00:00:00')
+            ],
+            [
+                'Avg'       => 2,
```

Comment:
Verwende besser verschiedene Avg-Werte. Damit verhinderst du, dass beispielsweise aus Versehen der gleiche Tag dreimal genommen wird und tatsächlich alle Zahlen mit in die Berechnung fließen